### PR TITLE
Development setup with docker compose

### PR DESCRIPTION
**Pull request by manuhdez**
The application has been containerized with docker in order to launch a local version easily without extra setup.

Services configured:
- backend (The actual app running with nodejs v15.0)
- mongodb (To persist the application data)

### CODE REVIEW

**Comment by tedsecretsource** [comment]
`docker compose build` seems to work but `docker compose start` doesn't seem to do anything and `docker compose ps` shows that no containers are running, so, something is failing. I assume there are logs somewhere that would provide more info but meanwhile, docker desktop is rebuilding itself!

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  database:
+    image: mongo:5.0.0-rc1-focal
+    restart: always
+    ports:
+      - "27017:27017"
+    volumes:
+      - mongodb:/data/db
+
+volumes:
+  mongodb:
```
Comment (tedsecretsource) [inline docker-compose.yml:21]: Should this have a value, maybe just a dot?

**Review by tedsecretsource — CHANGES_REQUESTED** [review]
I'm not seeing where we do `yarn` to install dependencies (although I don't think that that's the problem I'm having).

File: docker-compose.yml (hunk 1 above)
Comment (manuhdez) [reply docker-compose.yml:21]: I read a couple of articles and it looks like you only need that reference but I haven't been able to properly test it.
I will be adding data into the database and test if the data persist within the container both with and without this reference.

**Comment by manuhdez** [comment]
> `docker compose build` seems to work but `docker compose start` doesn't seem to do anything and `docker compose ps` shows that no containers are running, so, something is failing. I assume there are logs somewhere that would provide more info but meanwhile, docker desktop is rebuilding itself!

I think we might need to stick to `docker-compose up --build` for the first build and `docker-compose up -d` to launch the containers.

File: docker-compose.yml (hunk 1 above)
Comment (manuhdez) [reply docker-compose.yml:21]: As you can see [here](https://stackoverflow.com/a/64020164/10699118), declaring the volume like this is equivalent to run the command `docker volume create mongodb`, which lets docker to manage the volume automatically.

**Pull request merged**